src/BuildToolsUnitTests/DataContractAnalyzerTests.cs

```diff
@@ -172,7 +202,40 @@ public class Foo
 }");
             var diag = Assert.Single(diagnostics, x => x.Descriptor == DataContractAnalyzer.DuplicateFieldNumber);
             Assert.Equal(DiagnosticSeverity.Error, diag.Severity);
-            Assert.Equal($"The specified field number 2 is duplicated; field numbers must be unique between all declared members and includes on a single type.", diag.GetMessage(CultureInfo.InvariantCulture));
+            Assert.Equal(
+                $"The specified field number 2 is duplicated; field numbers must be unique between all declared members and includes on a single type.",
+                diag.GetMessage(CultureInfo.InvariantCulture));
+        }
+
+        [Fact]
+        public async Task ReportsDuplicateFieldBetweenFieldsWithDifferentAttributes()
+        {
+            var diagnostics = await AnalyzeAsync(@"
+using ProtoBuf;
+using System.Runtime.Serialization;
+[ProtoContract]
+public class Foo
+{
+    [ProtoMember(1)]
+    public int A {get;set;}
+
+    [DataMember(Order = 2)]
```

Comment:
This is tricky; IIRC the serializer will ignore this member, so... is it actually a conflict?